docs/recipes/decouplingReducers.md

```diff
@@ -0,0 +1,141 @@
+# Decoupling reducers
+
+Rematch and more generally Redux, encourage to keep a flat state to solve several [issues](https://redux.js.org/recipes/structuring-reducers/normalizing-state-shape).
+Each model should be considered as a separate entity and should only be updated by its reducer, this is why Rematch doesn't allow nested reducers.
+
+This being said, there is some case where nested reducer can be a good thing.
+In this recipe, we will see how to decoupling reducers that tends to grow too much.
+
+Let's start with an example. A common pattern in Redux, when we want to store an ordered list, is "allIds, byId". Store  the entities as
+an array requires to iterate over the array to find our target but store the entities as an object doesn't let us to keep 
+an order. So let's use both: 
```

Comment:
"Storing the entities as an array requires an additional cost of iterating over the array to find our target. On the other hand, storing the entities as an object doesn't preserve the order of items"